### PR TITLE
Permission checking

### PR DESCRIPTION
**Pull request by nishadg246**
Pulled out permission checking status into
a singleton class so both the activity and
fragment could access it easily

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 1]
```diff
@@ -180,13 +180,12 @@ public synchronized void disconnect() {
...
       stopScanningDisplay();
+      notifyChangeOnUiThread();
     }
   }
 
-  public static NearbyBeaconsFragment newInstance(MainActivity activity) {
-    NearbyBeaconsFragment fragment = new NearbyBeaconsFragment();
-    fragment.mMainActivity = activity;
-    return fragment;
+  public static NearbyBeaconsFragment newInstance() {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: just get rid of this method

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 2]
```diff
@@ -222,18 +221,18 @@ public View onCreateView(LayoutInflater layoutInflater, ViewGroup container,
 
   @Override
   public void onResume() {
+    mPermissionCheck = PermissionCheck.getInstance();
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: Why not in onCreate?

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 3]
```diff
@@ -222,18 +221,18 @@ public View onCreateView(LayoutInflater layoutInflater, ViewGroup container,
...
     super.onResume();
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: If you keep the previous line, onresume needs to come first

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 4]
```diff
@@ -78,7 +78,7 @@
   private boolean mSecondScanComplete;
   private boolean mFirstTime;
   private DiscoveryServiceConnection mDiscoveryServiceConnection;
-  private MainActivity mMainActivity;
+  private PermissionCheck mPermissionCheck;
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: I don't think we need a member just get the instance when you need it.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MainActivity.java [hunk 5]
```diff
@@ -42,12 +42,14 @@
   private static final int REQUEST_LOCATION = 1;
   private static final String NEARBY_BEACONS_FRAGMENT_TAG = "NearbyBeaconsFragmentTag";
   private boolean mCheckingPermissions = false;
+  private PermissionCheck mPermissionCheck;
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MainActivity.java]: Don't bother with the member.  Just get the singleton when you need it.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 6]
```diff
@@ -180,15 +179,10 @@ public synchronized void disconnect() {
       mDiscoveryService = null;
       getActivity().unbindService(this);
       stopScanningDisplay();
+      notifyChangeOnUiThread();
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: Why is this needed?

**Comment by nishadg246** [comment]
@cco3

**Comment by cco3** [comment]
LGTM, but you are going to need to rebase it.

**Comment by nondebug** [comment]
lgtm
